client/influxdb.go

```diff
@@ -178,19 +197,43 @@ func (c *Client) Query(q Query) (*Response, error) {
 	defer resp.Body.Close()
 
 	var response Response
-	dec := json.NewDecoder(resp.Body)
-	dec.UseNumber()
-	decErr := dec.Decode(&response)
+	if q.Chunked {
+		cr := NewChunkedResponse(resp.Body)
+		for {
+			r, err := cr.NextResponse()
+			if err != nil {
+				// Ignore EOF errors if we got an invalid status code.
+				if err == io.EOF && resp.StatusCode != http.StatusOK {
```

Comment:
`NextResponse` can't return `io.EOF` so I don't think you need to check for it here.